client/jetpack-connect/auth-form-header.jsx

```diff
@@ -139,8 +138,8 @@ export class AuthFormHeader extends Component {
 		const safeIconUrl = siteIcon ? safeImageUrl( siteIcon ) : false;
 		const icon = safeIconUrl ? { img: safeIconUrl } : false;
 		const url = decodeEntities( homeUrl );
-		const parsedUrl = urlModule.parse( url );
-		const path = parsedUrl.path === '/' ? '' : parsedUrl.path;
+		const parsedUrl = new URL( url );
```

Comment:
Can `url` ever be anything other than an absolute URL, with a protocol and host? If so, then this won't work, as the standard `URL` API that's built into the browser can only handle absolute URLs.

You only seem to need the pathname for this, so it may still be doable without using `lib/url`. You'll need to provide a base URL for URL to fill in the gaps:

```js
const parsedUrl = new URL( url, 'http://example.com' );
```

This way you'll get `http://example.com/my/path` if `url` is simply `/my/path` or `//example.com/my/path` (no protocol).

Of course, that could still fail if `url` is an invalid URL, and depending on your goals you may want to let that exception propagate, or handle it differently somehow.

Welcome to the wonderful world of `URL`, @tyxla 😄

Reply:
Thanks for clarifying that @sgomes 😉 URLs are hard, aren't they? 😄 

> Can url ever be anything other than an absolute URL, with a protocol and host? I

Luckily, no. Jetpack Connect does not support non-absolute URLs, so I think we are safe here. 

> You only seem to need the pathname for this,

Well, we're also using the `host` on the following line.

Anyway, given the context I'm providing, do you have any other concerns regarding this change?

Reply:
Also, bear in mind that `url`'s `path` is not the same as `URL`'s `pathname`. `path` also includes the query string if it's present:

```js
import url from 'url';
const example = 'http://example.com/path?a=b#c';
console.log( url.parse( example ).path ); // -> /path?a=b
```

```js
const example = 'http://example.com/path?a=b#c';
console.log( new URL( example ).pathname ); // -> /path
```

Reply:
@tyxla Sorry, only saw your reply after I had posted my reply to myself 🙂 

Other than the new issue I pointed out, this should be fine. Another fun fact: `host` !== `hostname`; the former includes port if it's present, while the latter does not. I expect that not to matter much for this use-case 🙂

Reply:
Good call 👍 

But I'd also say that we're fixing this case, then. A site's home URL shouldn't contain any query strings, so if we're omitting them here, that's even an improvement to what we had before.

Reply:
We might as well have chatted in Slack 😆 

And yeah, we're intentionally preserving the port, although it's usage is really an edge case for most of the sites.

Reply:
Perfect, thanks for confirming everything! 👍 And as for Slack vs issues, well, maybe someone will find this discussion useful at some point 😄